### PR TITLE
task/free_draw/view_model_drawer

### PR DESCRIPTION
**Pull request by teamagam5**
Feature basically works.

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+      sLogger.v("No action to undo");
+    }
+  }
+
+  public void setColor(String color) {
+    mSymbol = createSymbol(color);
+  }
+
+  public void switchMode() {
+    toggle();
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java]: toggle what?

File: app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    mIsDrawingMode = !mIsDrawingMode;
+  }
+
+  private interface Command {
+    void execute();
+
+    void undo();
+  }
+
+  private abstract class MapActionCommand implements Command {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java]: It seems like this isn't really a classic use of an inheritance as the inheriting classes don't get any "benefit" (/behavior) if they inherit from this base class unless they specifically call these newly introduced protected methods.
In this case, I'd rather use composition here (meaning, MapActionCommand will not implement command, but it'll be used to implement the command interface in the inheriting classes).
The main benefit is leaving the actual commands "free" to inherit, if needed, from a more meaningful base-class (one that will add some behavior to its descendants)

File: domain/src/main/java/com/teamagam/gimelgimel/domain/map/entities/geometries/PointGeometry.java [hunk 3]
```diff
@@ -3,54 +3,58 @@
...
 
-  /**
-   * constructor with altitude
-   */
   public PointGeometry(double latitude, double longitude, double altitude) {
-    this.mLatitude = latitude;
-    this.mLongitude = longitude;
-    this.mAltitude = altitude;
-    this.mHasAltitude = true;
+    this(latitude, longitude, true, altitude);
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/map/entities/geometries/PointGeometry.java:21]: well done

File: app/src/test/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawerTest.java [hunk 4]
```diff
@@ -0,0 +1,600 @@
+package com.teamagam.gimelgimel.app.map.actions.freedraw;
```
Comment (BarAgam) [inline app/src/test/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawerTest.java:1]: Duuuuuuude, this class is so big github wouldn't show it to me by default.

**Review by BarAgam — APPROVED** [review]
You'll need to rebase onto develop.
Once you do that, you can refactor to reuse some symbol choosing logic shared with the other drawing features (measure/polyX/dynamic layer)

File: app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java (hunk 1 above)
Comment (teamagam5) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java]: I inlined it

File: app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java (hunk 2 above)
Comment (teamagam5) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java]: I'm not sure I get what u mean by saying they don't share behaviour, but anyway I agree it was an unnecessary complication. I got rid of `MapActionCommand`, moved the 2 operations to the outer (main) class, and used them within the 2 `Command` implementations.

**Pull request merged**